Update roles for kube-agent chart in guides/documentation to be 'kube,app,discovery'

AntonAM opened the pull request:
New feature of Kuberentes app discovery is controlled by roles provided to helm chart (it's automatically enabled if discovery and app roles are present). We want users to not have problems with feature discovery of Kubernetes app discovery, so we're making sure by default appropriate guides/docs tell users to enable it by setting roles to be 'kube,app,discovery'.

AntonAM committed:
Update default roles for kube-agent chart in guides/documentation to be 'kube,app,discovery'

We want users to not have problems with feature discovery of Kuberentes app discovery,
so we're making sure by default our guides tell users to enable it.

tigrato commented on docs/pages/api/rbac.mdx:
````diff
@@ -344,6 +344,7 @@ requested earlier:
 ```code
 $ helm install teleport-agent teleport/teleport-kube-agent \
   --set kubeClusterName=minikube \
+  --set roles="kube\,app\,discovery"
````
```suggestion
  --set roles="kube\,app\,discovery" \
```

AntonAM replied:
🤦

tigrato commented on docs/pages/kubernetes-access/manage-access/rbac.mdx:
````diff
@@ -226,6 +226,7 @@ requested earlier:
 ```code
 $ helm install teleport-agent teleport/teleport-kube-agent \
   --set kubeClusterName=minikube \
+  --set roles="kube\,app\,discovery"
````
```suggestion
  --set roles="kube\,app\,discovery" \
```

tigrato commented on docs/pages/kubernetes-access/register-clusters/iam-joining.mdx:
```diff
@@ -249,6 +249,7 @@ $ CLUSTER=iam-cluster
 $ PROXY=tele.example.com:443
 $ helm install teleport-agent teleport/teleport-kube-agent \
   --set kubeClusterName=${CLUSTER?} \
+  --set roles="kube\,app\,discovery"
```
```suggestion
  --set roles="kube\,app\,discovery" \
```

tigrato commented on examples/chart/index.html:
```diff
@@ -28,6 +28,7 @@ <h4>See our <a href="https://goteleport.com/docs/kubernetes-access/helm/guides/"
       helm install ${RELEASE_NAME?} teleport/teleport-kube-agent \
         --create-namespace \
         --namespace ${NAMESPACE?} \
+        --set roles="kube\,app\,discovery"
```
```suggestion
        --set roles="kube\,app\,discovery" \
```

tigrato commented on tool/tctl/common/kube_command.go:
```diff
@@ -122,6 +122,7 @@ helm repo update
 
 > helm install teleport-agent teleport/teleport-kube-agent \
   --set kubeClusterName=cluster ` + "`" + `# Change kubeClusterName variable to your preferred name.` + "`" + ` \
+  --set roles="{{.set_roles}}"
```
```suggestion
  --set roles="{{.set_roles}}" \
```

tigrato commented on tool/tctl/common/token_command.go:
```diff
@@ -257,11 +257,13 @@ func (c *TokensCommand) Add(ctx context.Context, client auth.ClientI) error {
 		if len(proxies) == 0 {
 			return trace.NotFound("cluster has no proxies")
 		}
+		setRoles := strings.ToLower(strings.Join(roles.StringSlice(), "\\,"))
```
Should we enable `app & discovery` if type = kube?

AntonAM replied:
It would be impossible for user to add token just for `kube` roles then, do you think this is ok?

tigrato replied:
It's the same behavior we have in adding new resources via web UI.

AntonAM replied:
Added in https://github.com/gravitational/teleport/pull/30291/commits/e3ce9ac0c6b74ed80f7e9710f5f2f6d105f9b46a

AntonAM committed:
Add missing backslash

ptgott commented:
@AntonAM Are we backporting this?

AntonAM commented:
@ptgott no, kube app discovery will be released just from 14.

marcoandredinis commented on tool/tctl/common/token_command.go:
```diff
@@ -257,11 +257,13 @@ func (c *TokensCommand) Add(ctx context.Context, client auth.ClientI) error {
 		if len(proxies) == 0 {
 			return trace.NotFound("cluster has no proxies")
 		}
+		setRoles := strings.ToLower(strings.Join(roles.StringSlice(), "\\,"))
```
Can we simplify this and remove the extra quotes?
Roles are `[a-z]` so we should be able to just join them by `,`
Can we please add a test for this?

AntonAM replied:
Because of the way Helm parses values it should be in the form `--set roles="kube\,app\,discovery"`, it seems nothing else works. I've added a test checking this.

AntonAM committed:
Add enabling of App and Discovery roles when generating token for Kube

We want users to not have problems with enabling Kuberneetes App discovery by default.

AntonAM committed:
Add test checking that we provide correct '--set roles' text in command print out.
